src/goto-cc/gcc_mode.cpp

```diff
@@ -688,7 +689,9 @@ int gcc_modet::doit()
 
           std::string new_name=
             get_base_name(arg_it->arg, true)+new_suffix;
-          std::string dest=temp_dir(new_name);
+
+          temp_dirs.emplace_front("goto-cc-XXXXXX");
+          std::string dest = temp_dirs.front()(new_name);
```

Comment:
If we’re already doing temp dirs and what not, couldn’t we create proper temp files for the destinations of these things instead of maintaining a list of temp directories?

Reply:
Ah yes there's already `temporary_filet` to create a temporary file (and it also handles the copying and moving correctly).

Reply:
Ok, switched this back to using temporary directories, as the name of the preprocessed file is used at various places (such as when deducing the name of the object file when goto-cc is invoked with a single source file, i.e., `goto-cc -c test.c`).